ddtrace/contrib/django/patch.py

```diff
@@ -497,6 +512,13 @@ def _patch(django):
         import django.views.generic.base
     trace_utils.wrap(django, "views.generic.base.View.as_view", traced_as_view(django))
 
+    try:
+        import channels.routing
+
+        trace_utils.wrap(channels.routing, "URLRouter.__init__", unwrap_views)
+    except ImportError:
+        pass  # channels is not installed
```

Comment:
Does this work on all versions of `channels`?

Reply:
Yup, the file channels/routing.py exists in all versions of channels (v0.X, v1, v2, and v3). However URLRouter only exists in channels v2.0 to latest

Reply:
can we get the channels version? should we only patch this if we know we are on channels >= 2.0?

Reply:
Actually this fix is only required for channels v3. ASGI3 is not supported in channels 2